decoder/vaapidecoder_h264.cpp

```diff
@@ -1783,6 +1783,12 @@ YamiStatus VaapiDecoderH264::decodeNalu(NalUnit* nalu)
     return status;
 }
 
+void VaapiDecoderH264::flush(void)
+{
+    m_dpb.flush();
```

Comment:
could you move all things between https://github.com/01org/libyami/blob/apache/decoder/vaapidecoder_h264.cpp#L1825 and  https://github.com/01org/libyami/blob/apache/decoder/vaapidecoder_h264.cpp#L1830 to flush? and call flush in L1825